### ISSUE TITLE
IndexOutOfRangeException Error

### ISSUE THREAD

**Issue by spooky-sooshi** [Ishigh1/RecursiveCraft#2]
Getting an IndexOutOfRangeException in the chat box whenever I try recursively crafting any item in the Magic Storage interface. No crashes, it just doesn't craft the item and sends out an error.

Full Message
```
System.IndexOutOfRangeException: Index was outside the bounds of the array.
   at Terraria.ModLoader.ItemLoader.OnCraft(Item item, Recipe recipe)
   at RecursiveCraft.CompoundRecipe.OnCraft(Item item) in RecursiveCraft\CompoundRecipe.cs:line 100
   at Terraria.ModLoader.RecipeHooks.OnCraft(Item item, Recipe recipe)
   at MagicStorageExtra.CraftingGUI.TryCraft() in MagicStorageExtra\CraftingGUI.cs:line 1533
   at MagicStorageExtra.CraftingGUI.UpdateCraftButton() in MagicStorageExtra\CraftingGUI.cs:line 711
   at MagicStorageExtra.CraftingGUI.Update(GameTime gameTime) in MagicStorageExtra\CraftingGUI.cs:line 432
   at Terraria.ModLoader.ModHooks.PostUpdateInput()
   at Terraria.Main.DoUpdate_HandleInput()
   at Terraria.Main.DoUpdate(GameTime gameTime)
   at DMD<Terraria.Main::Update>(Main , GameTime )
   at Microsoft.Xna.Framework.Game.Tick()
   at Microsoft.Xna.Framework.SDL2_FNAPlatform.RunLoop(Game game)
   at Microsoft.Xna.Framework.Game.Run()
   at Terraria.Program.LaunchGame_()
   at Terraria.Program.LaunchGame(String[] args, Boolean monoArgs)
   at Terraria.FNALaunch.Main(String[] args)
```

Using v1.4.1

**Comment by spooky-sooshi** [comment]
> It doesn't happen on my side, what mods are you using ?

Modlist:
```
"AAMod"
"AccessorySlotsPlus"
"AlchemistNPCLite"
"Antisocial"
"ApothTestMod"
"AutoReroll"
"AutoTrash"
"BestModifiers"
"BossChecklist"
"CalamityMod"
"CalamityRecipeAddon"
"CalValEX"
"CataclysmMod"
"DoubleOreDrop"
"DualWielding"
"ElementsAwoken"
"ItemsBuffs"
"ExtensibleInventory"
"Fargowiltas"
"FargowiltasSouls"
"FargowiltasSoulsDLC"
"FasterBuilding"
"FasterDamages"
"FasterPickaxes"
"FKBossHealthBar"
"GoldensMisc"
"HamstarHelpers"
"imkSushisMod"
"imkSushisModOldRecipesEnabler"
"IndustrialPickaxes"
"InstantRespawn"
"LithosArmory"
"PboneUtils"
"MagicalThings"
"MaxStackExtra"
"MoreAccessories"
"MoreEmblems"
"NoMoreTombs"
"OmniSwing"
"RecipeBrowser"
"Redemption"
"ReducedGrinding"
"SacredTools"
"ShopExpander"
"SmartDoors"
"SmoothLighting"
"SpiritMod"
"ThoriamityConvergence"
"ThoriumMod"
"UpgradedAccessories"
"VeinMiner"
"WingSlot"
"WMITF"
"ZLegendarys"
"SubworldLibrary"
"Consolaria"
"Intrinsics"
"MountedMagicMirrors"
"NoFavoriteTooltips"
"UpgradeEquipment"
"Split"
"MrPlagueRaces"
"SteelsRace"
"BetterEmblems"
"_ColoredDamageTypes"
"pinkymod"
"LansUnlimitedBuffSlots"
"TianicCustomRaces"
"StopDying"
"TysSimplifiedSummoning"
"OrchidMod"
"OneStopNPCShop"
"NoFishTimer"
"Wireless"
"CensusSupport"
"Census"
"BetterManaRegenFixed"
"BetterBuffs"
"DirectionalMelee"
"SolutionsMod"
"ShaderLib"
"ItemCustomizer"
"MagicStorageExtra"
"MechTransfer"
"DiscordRP"
"NameCorrector"
"NokiaMod"
"CombinedExpertItems"
"Gerdsmod"
"UltimatePotionsMod"
"BaseMod"
"GRealm"
"TrueEndless"
"MinigunAndOpBullets"
"DupeFrame"
"Terrarchitect"
"ZenithsItems"
"ObamaPrismMod"
"EndlessCrafting"
"BetterAutosave"
"ItemModifier"
"GadgetGalore"
"VipixToolBox"
"Polarities"
"UsefulNPCs"
"PicklockNPC"
"MiniBossNPC"
"GrabBagOpener"
"DisableCorruptionSpread"
"ImprovedPillars"
"ChatSource"
"BuoyancyMod"
"AutoPotion"
"Ultranium"
```

It's an error on my side, instead of the mod's so I might just close this issue to be honest. The sheer number of mods might be the cause for the error.

**Comment by Ishigh1** [comment]
Ok, so it's Mod of Redemption that's causing the issue, we'll investigate about it

**Comment by Ishigh1** [comment]
Fixed in Recursive Craft 1.6
